src/components/FeedPage.vue

```diff
@@ -85,6 +91,9 @@ export default {
                 this.loadMoreVideos();
             }
         },
+        onUpdateWatched() {
+            if (this.videos.length > 0) this.updateWatched(this.videos);
+        },
```

Comment:
We don't need to update all videos if just one video was watched!

Reply:
I agree! However - as far as I can tell - `updateWatched` alters the array passed to it so `this.videos` needs to be used. Maybe there's something I can do here, constructing a smaller array with indexes or something, let me look into it.

Reply:
Well that was easy lol